#! /bin/sh /usr/share/dpatch/dpatch-run
## array-out-of-bounds.dpatch by Thomas Weber <thomas.weber.mail@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes a too large index in the Sparse code, thanks to Atsushi Kajita
## DP: http://www.cae.wisc.edu/pipermail/bug-octave/2006-June/000428.html

@DPATCH@
diff -urNad octave2.9-2.9.6~/liboctave/Sparse.cc octave2.9-2.9.6/liboctave/Sparse.cc
--- octave2.9-2.9.6~/liboctave/Sparse.cc	2006-05-23 08:05:14.000000000 +0200
+++ octave2.9-2.9.6/liboctave/Sparse.cc	2006-06-29 19:51:41.000000000 +0200
@@ -59,7 +59,7 @@
 
       // Ok, If we've gotten here, we're in trouble.. Have to create a 
       // new element in the sparse array. This' gonna be slow!!!
-      if (c[ncols+1] == nzmx)
+      if (c[ncols] == nzmx)
 	{
 	  (*current_liboctave_error_handler)
 	    ("Sparse::SparseRep::elem (octave_idx_type, octave_idx_type): sparse matrix filled");
